### PR TITLE
Use action link in save-in-progress component to start form

### PR DESCRIPTION
**Pull request by Mottie**
## Description

The design for our Notice of Disagreement form is using the new action link within the `SaveInProgressIntro` component that starts the form.

In this PR, the `FormStartControls` component is modified to render an action link when a `useActionLink` property is passed to it from the `SaveInProgressIntro` component instead of rendering a `ProgressButton`. The `SaveInProgressIntro` relays this property from the specific form's Introduction page. So this change will only apply to:
- Forms that specifically set a `useActionLink` property.
  ```html
  <SaveInProgressIntro
    testActionLink
    prefillEnabled={formConfig.prefillEnabled}
    messages={formConfig.savedFormMessages}
    pageList={pageList}
    startText="Start the Application"
  />
  ```
- The start application button when the user is signed in

Related:
- Ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/22699
- Design: https://vsateams.invisionapp.com/share/8Y10I6K7DU9R#/screens/447558866

## Testing done

Added unit test

## Screenshots

Screen Shot 2021-04-12 at 1 21 00 PM

## Acceptance criteria
- [x] Use action links to start the form per setting

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/save-in-progress/FormStartControls.jsx [hunk 1]
```diff
@@ -132,12 +132,24 @@ class FormStartControls extends React.Component {
         </div>
       );
     }
-
-    return (
+    const startText = this.props.startText || 'Get Started';
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/FormStartControls.jsx:135]: Can `Get Started` be moved into `defaultProps`?

Comment (Mottie) [reply src/platform/forms/save-in-progress/FormStartControls.jsx]: Done

**Comment by cvalarida** [comment]
I'm all for making this an action link, but I don't love the idea of adding yet more configuration. Let's get design consensus from the design system council and make it happen across all forms by default (without that extra `useActionLink` prop).

**Comment by Mottie** [comment]
> make it happen across all forms by default

I can do that. Let's discuss this on Friday :D

**Comment by Mottie** [comment]
Changed to `testActionLink`

**Review by cvalarida — APPROVED** [review]
Per the decision from the design system council today to run an experiment to ensure the action link tests well enough to move to it on all forms, this look great. :+1:

**Pull request merged**